packages/darklang/prettyPrinter/common.dark

```diff
@@ -16,4 +16,21 @@ module Darklang =
     let sign (s: PACKAGE.Darklang.Stdlib.Sign) : String =
       match s with
       | Positive -> ""
-      | Negative -> "-"
\ No newline at end of file
+      | Negative -> "-"
+
+    let escapeSpecialCharacters (s: String) : String =
+      s
+      |> String.replaceAll "\n" "\\n"
+      |> String.replaceAll "\t" "\\t"
+      |> String.replaceAll "\r" "\\r"
+
+    let processRemainder (remainder: String) : String =
+      let rem = $".{remainder}" |> Float.parse
+
+      match rem with
+      | Ok r ->
+        if r == 0.0 then
+          "0"
+        else
+          (r |> Float.toString) |> String.dropFirst 2
+      | Error e -> e
```

Comment:
Is it bad to return the error here?

Reply:
what would you expect to happen if you parse an invalid float?

Reply:
I expect an error or an exception to be thrown indicating that parsing an invalid float isn't possible
The issue is if I have something like 
let z = 0.abc
let a = 3.14abc
in the pretty printer is is displayed as 
let z = 0
let a = 0

Reply:
The difference here, if I understand it correctly, is that this is pretty printing EFloats, where it should not be possible to have invalid floats present.

Reply:
Currently, when we send an invalid float like `1.b`, it doesn't reach the `EFloat` branch; instead, it is considered as an `EInt`, and as a result, returns 0. I don't believe this issue is related to the pretty printer. I've added some tests in 'Int.test' for this scenario (i.e., 'Int.toString 1.ab = "0"' passes)